lib/foreman_maintain/cli/backup_command.rb

```diff
@@ -0,0 +1,156 @@
+module ForemanMaintain
+  module Cli
+    module BackupCommon
+      def self.included(klass)
+        klass.extend ClassMethods
+      end
+
+      def backup_dir
+        @backup_dir ||= preserve_directory? ? backup_root_dir : backup_subdir
+      end
+
+      def timestamp
+        DateTime.now.strftime('%Y-%m-%d-%H-%M-%S')
+      end
+
+      def backup_subdir
+        File.join(backup_root_dir, 'foreman-backup-' + timestamp)
```

Comment:
On Satellite, this is currently called `satellite-backup-<timestamp>`, which means this will be different. Can we check for downstream and change dir name as needed?

Reply:
Also would be nice if it checked for katello feature and called the backup `katello-backup` as well, since this is what the current backups are called upstream.

Reply:
I've also added proxy and capsule